src/fft-api/facility/fftFacilityService.ts

```diff
@@ -29,26 +29,34 @@ export class FftFacilityService {
 
   constructor(private readonly apiClient: FftApiClient) {}
 
-  public async getFacilityId(tenantFacilityId: string): Promise<string> {
+  public async getFacilityId(tenantFacilityId: string, lenient = false): Promise<string | undefined> {
     if (FftFacilityService.facilityCache.has(tenantFacilityId)) {
       return FftFacilityService.facilityCache.get(tenantFacilityId) as string;
     }
 
     const strippedFacilities = await this.apiClient.get<StrippedFacilities>(this.PATH, { tenantFacilityId });
-    if (!strippedFacilities.facilities?.length) {
-      this.logger.error(`Did not find facility with tenantFacilityId '${tenantFacilityId}'`);
-      throw new Error(`Did not find facility with tenantFacilityId '${tenantFacilityId}'`);
-    }
+    const length = strippedFacilities.facilities?.length || 0;
```

Comment:
Would be a nullcheck after calling the API be better than optional chaining every time we access that property?

Reply:
refactored the if cases